OpenSourcePolitics/decidim-module-extra_user_fields
# feat: Add underage confirmation in extra_user_fields

#### :tophat: Description

This feature adds an underage confirmation in extra user fields that is directly linked to the date of birth if it is activated with a configurable minimal age in the range of 15 to 21 , 

#### :pushpin: Related Issues
*Link your PR to an issue*
- [Notion card](https://www.notion.so/opensourcepolitics/Toulouse-0-27-Ajout-d-une-option-autorisation-parentale-au-module-extra-sign-up-fields-97dbaea5c88b4fddbce1719ae66343e4?pvs=4)

#### Testing
*Describe the best way to test or validate your PR.*

Example:
* Log in as admin
* Access BackOffice and access to EUF fields

##### 1st test : 
* Enable only the underage field
* Try to create an account 
* Check that there are two mails sent, the one for the user to confirm his account and the second one to the statutory representative that was mentioned
*
##### 2nd test : 
* Enable the underage field with the date of birth and configure the minimum age required
* Check that when you come to the page there is not any checkbox displayed
* Try to set a birth date and register your user
* If this is below it should appear and then be ticked automatically
* Otherwise it should not be displayed at all
* Try to register your user

##### Some use cases to test : 
* Try to change the birth date from an invalid to a valid one
* Try to untick the checkbox and see what happens (should not create the user) with an "invalid" birth date

#### Tasks
- [x] Add the field
- [x] Add some specs
- [x] Add a JS file to handle the signup field combinated to the date of birth one
- [x] Modify the user
- [x] Add some verifications and fix some tests that weren't done
- [x] Create a mailer to send the mail to the statutory representative
- [x] Backport the mail format from Toulouse
